src/domain/service/noteSettings.ts

```diff
@@ -20,14 +22,47 @@ export default class NoteSettingsService {
   /**
    * Note Settings service constructor
    *
-   * @param noteSettingsrepository - note settings repository
+   * @param noteSettingsRepository - note settings repository
    * @param teamRepository - team repository
    */
-  constructor(noteSettingsrepository: NoteSettingsRepository, teamRepository: TeamRepository) {
-    this.noteSettingsRepository = noteSettingsrepository;
+  constructor(noteSettingsRepository: NoteSettingsRepository, teamRepository: TeamRepository) {
+    this.noteSettingsRepository = noteSettingsRepository;
     this.teamRepository = teamRepository;
   }
 
+  /**
+   * Add user to the team by invitation hash
+   *
+   * @param invitationHash - hash for joining to the team
+   * @param userId - user to add
+   */
+  public async addUserToTeamByInvitationHash(invitationHash: InvitationHash, userId: User['id']): Promise<TeamMember | null> {
+    const defaultUserRole = MemberRole.read;
+    const noteSettings = await this.noteSettingsRepository.getNoteSettingsByInvitationHash(invitationHash);
+
+    /**
+     * Check does invitation hash is valid
+     */
+    if (noteSettings === null) {
+      throw new Error(`Note with invitation ${invitationHash} does not exists`);
```

Comment:
maybe just `Wrong invitation`? to not to expose technical implementation details for api user

Reply:
changed message